File: Fortran/File.f90
subroutine file

 character home*128
 character md1*128
 character md2*128
 character md3*128

 print '(/,"これからファイル操作を試します",/)'

 print '("> カレントディレクトリをホームに変更",/)'
 call getenv("HOME",home)
 call chdir(home)

 ! ファイル/フォルダ/シンボリックリンクの作成

 print '("> Emptyという空フォルダを作成")'
 call system("mkdir Empty")
 ! つまり,Fortranではフォルダを作成できない

 print '("> Blankという空ファイルを作成")'
 open(unit=11,file="Blank")
 close(11)

 print '("> フォルダEmptyの中にファイルBlankのシンボリックリンクSymlinkを作成",/)'
 call symlnk("Blank","Empty/Symlink")

 ! 書込み

 print '("> Untitled.mdというMarkdownファイルを作成して書込み",/)'
 open(unit=12,file="Untitled.md",action="write")
 write(12,'("# Header 1")')
 close(12)

 ! 移動/名称変更

 print '("> フォルダEmptyをPackageに名称変更")'
 call rename("Empty","Package")
 print '("> Packageフォルダ内のSymlinkファイルをAliasに名称変更")'
 call rename("Package/Symlink","Package/Alias")
 print '("> Untitled.mdを移動して,名称変更")'
 call rename("Untitled.md","Package/Headers.md")

 ! 追記

 print '("> Markdownファイルに追記")'
 open(unit=13,file="Package/Headers.md",action="write",position="append")
 write(13,'("## Header 2",/,"### Header 3")')
 close(13)

 ! 読込み

 print '("> Markdownファイルを読込み")'
 open(unit=14,file="Package/Headers.md",action="read")
 read(14,'(a,a,a)') md1,md2,md3
 close(14)
 print '(a,a,a)',md1,md2,md3

 ! ファイル/フォルダの削除

 print '("> ファイルBlankを削除")'
 call unlink("Blank")

 print '("> シンボリックリンクAliasを削除")'
 call unlink("Package/Alias")

end subroutine file

! Fortranでは,ファイルハンドラとして装置番号を指定する
! 装置番号: 1~99 の任意の整数
! 但し,多くのシステムで以下の装置番号は予約されているため,利用できない。
!   5: 標準入力
!   6: 標準出力
!   0: 標準エラー出力

! open関数のstatusに指定できるのは,以下のオプション
!   'old' : ファイルが指定した場所に存在していなければならない。
!   'new' : ファイルが指定した場所には存在してはならない。
!   'replace' : ファイルが指定した場所に存在していれば新規空ファイルに置換され,存在していなければ新規空ファイルが作成される。

File: Go/File.go
package main

import "fmt"
import "os"
import "io/ioutil"

func File() {

	fmt.Println("\r\nこれからファイル操作を試します\r\n")

	// ファイル操作系のメソッドは基本的にエラーを発生させる可能性があり,返値でエラーを受け取っている。

	var e error

	fmt.Println("> カレントディレクトリをホームに変更\r\n")
	home,e := os.UserHomeDir()
	errorHandle(e)
	os.Chdir(home)

	// ファイル/フォルダ/シンボリックリンクの作成

	fmt.Println("> Emptyという空フォルダを作成")
	errorHandle(   os.Mkdir("Empty",0744)   )
		// Mkdirではなぜか7にしないとディレクトリへのアクセス権を失ってしまう

	fmt.Println("> Blankという空ファイルを作成")
	io1,e := os.Create("Blank")
	errorHandle(e)
	defer io1.Close()

	fmt.Println("> フォルダEmptyの中にファイルBlankのシンボリックリンクSymlinkを作成\r\n")
	errorHandle(   os.Symlink("Blank","Empty/Symlink")   )
		// os.Link("Blank","Symlink") とすると,ハードリンクが作成される

	fmt.Println("> Untitled.mdというMarkdownファイルを作成して書込み\r\n")
	io2,e := os.OpenFile("Untitled.md",os.O_WRONLY|os.O_CREATE,0644)
	errorHandle(e)
	io2.WriteString("# Header 1")
	defer io2.Close()
	/*
		或いは
		ioutil.WriteFile("Untitled.md",[]byte("# Header 1"))
	*/

	// 移動/名称変更

	fmt.Println("> フォルダEmptyをPackageに名称変更")
	errorHandle(   os.Rename("Empty","Package")   )
	fmt.Println("> Packageフォルダ内のSymlinkファイルをAliasに名称変更")
	errorHandle(   os.Rename("Package/Symlink","Package/Alias")   )
	fmt.Println("> Untitled.mdを移動して,名称変更\r\n")
	errorHandle(   os.Rename("Untitled.md","Package/Headers.md")   )

	// 追記

	fmt.Println("> Markdownファイルに追記\r\n")
	io3,e := os.OpenFile("Package/Headers.md",os.O_WRONLY|os.O_APPEND,0644)
	errorHandle(e)
	io3.WriteString("\r\n## Header 2\r\n### Header 3")
	defer io3.Close()

	// 読込み

	fmt.Println("> Markdownファイルを読込み")
	bytearray,e := ioutil.ReadFile("Package/Headers.md")
	errorHandle(e)
	fmt.Println(string(bytearray))
	fmt.Print("\r\n")

	// 再帰的にフォルダ作成

	fmt.Println("> フォルダを一気に作成\r\n")
	errorHandle(   os.MkdirAll("Package/Module/Submodule/Item",0744)   )
		// os.Mkdirの代わりにos.MkdirAllを使うことで,作成するフォルダItemの上位フォルダModule,Submoduleが存在していなくても,同時に生成される

	// ファイル/フォルダの削除

	fmt.Println("> ファイルBlankを削除")
	os.Remove("Blank")

	fmt.Println("> シンボリックリンクAliasを削除")
	os.Remove("Package/Alias")

	fmt.Println("> フォルダModuleを削除\r\n")
	os.Remove("Package/Module")
		// 空のフォルダはos.Removeで削除できる。空でない場合は削除できない。
		// os.RemoveAll を使えば,空でなくても削除できる

}

func errorHandle(e error) {
	if (e != nil) {
		fmt.Println("直前の操作においてエラーが発生しました")
		fmt.Println(e)
		os.Exit(1)
	}
}